### PR TITLE
Fix logic bugs of the transformer

### PR DESCRIPTION
**Pull request by thuangb**
+ Remove one seemingly unused text positional embedding
+ Output the logits starting from the end of segmentation position not at the start of image position

### CODE REVIEW

**Comment by thuangb** [comment]
hi @dome272, you should check these two errors. The second error is serious, because we must sample from the end of segmentation position
